assets/js/modules/analytics/components/common/ProfileSelect.js

```diff
@@ -116,11 +116,12 @@ export default function ProfileSelect( { hasModuleAccess } ) {
 		);
 	}
 
+	const displayProfiles = [ ...profiles ];
 	if (
 		! ga4ReportingEnabled &&
-		! profiles.some( ( profile ) => profile.id === PROFILE_CREATE )
+		! displayProfiles.some( ( profile ) => profile.id === PROFILE_CREATE )
 	) {
-		profiles.push( {
+		displayProfiles.push( {
```

Comment:
Hmm, although this works, I'd suggest a slightly different approach. How about instead of _always_ duplicating `profiles`, instead we assign `displayProfiles` using `let`, and then replace it instead of using `push()`? That means we clearly indicate it's a mutable variable at the point where it's defined, and means we don't unnecessarily duplicated an array of every render.

In other words:

```js
	let displayProfiles = profiles;
	if (
		! ga4ReportingEnabled &&
		! displayProfiles.some( ( profile ) => profile.id === PROFILE_CREATE )
	) {
		displayProfiles = [ ... profiles, {
		// ... etc.
```

Reply:
Updated ✅